src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -818,6 +818,35 @@ void QgsLayoutItemLegend::onAtlasEnded()
   updateFilterByMap();
 }
 
+QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
+{
+  QgsExpressionContext context = QgsLayoutItem::createExpressionContext();
+
+  //Can't utilize QgsExpressionContextUtils::mapSettingsScope as we don't always
+  //have a QgsMapSettings object available when the context is required, so we manually
+  //add the same variables here
+
+  // add the existing context of the linked map
+  if (mMap)
+  {
+    context.appendScope( mMap.createExpressionContext() );
+  }
+
+  QgsExpressionContextScope *scope = new QgsExpressionContextScope( tr( "Legend Settings" ) );
+
+  scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_title" ),title(),true);
+  scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "legend_settings" ),legendSettings(),true);
```

Comment:
I think you'll need to remove this one -- legendSettings() returns a QgsLegendSettings object, which cannot be convert to the QVariant type required by addVariable. Looks like you've already exposed the useful stuff from legend settings with the separate title/columns/etc variables.

Reply:
Honestly I wasn't too sure what I needed to expose, I feel like the scope for the modeltreelegendnode would be more useful and I assume is the next logical step.

Reply:
Just remove the legend_settings one -- the rest look useful, and in many ways its better to expose as much as possible, just in case someone NEEDS one of these!